src/plugins/ResizeReorder/ReorderCell.js

```diff
@@ -11,66 +11,71 @@
  */
 
 import React from 'react';
+import ReactDOM from 'react-dom';
 import joinClasses from '../../vendor_upstream/core/joinClasses';
 import cx from '../../vendor_upstream/stubs/cx';
 import FixedDataTableCellDefault from '../../FixedDataTableCellDefault';
-import { PluginContext } from '../../Context';
-import ReorderHandle from './ReorderHandle';
+import { FixedDataTableContext } from '../../FixedDataTableContext';
+import DragProxy from './DragProxy';
+import ExternalContextProvider from '../ExternalContextProvider';
 import _ from 'lodash';
 import ResizeCell from './ResizeCell';
 import PropTypes from 'prop-types';
 
 const BORDER_WIDTH = 1;
 
+/**
+ * HOC that enables reordering functionality by rendering a handle, which can be used to drag the cell around.
+ *
+ * While dragging takes place, this components rerenders a proxy component into a seperate React root, referred here as the "Drag Proxy".
+ * ReorderCell then hides itself, while the Drag Proxy becomes visible and is what the user actually sees and drags around.
+ *
+ * This ensures that even if ReorderCell unmounts (which can happen when the table scrolls far away), the Drag Proxy
+ * which is rendered in a separate React root is still alive, and hence doesn't break dragging functionality.
+ *
+ * Once dragging ends, the proxy is destroyed, and the original ReorderCell becomes visible again.
+ */
 class ReorderCell extends React.PureComponent {
   state = {
     isReordering: false,
-    displacement: 0,
   };
 
-  /**
-   * @param {string} columnKey
-   */
-  onColumnReorderStart = (columnKey) => {
-    this.setState({ isReordering: true });
-    this.props.onColumnReorderStart(columnKey);
-  };
+  dragContainer = null;
+  cellRef = React.createRef();
+  isMounted = false;
 
-  /**
-   * @param {number} displacement
-   */
-  translateCell = (displacement) => {
-    this.setState({ displacement });
-  };
+  componentDidMount() {
+    this.isMounted = true;
 
-  /**
-   * @param ev
-   * @param {string} ev.columnBefore
-   * @param {string} ev.columnAfter
-   * @param {string} ev.reorderColumn
-   */
-  onColumnReorderEnd = (ev) => {
-    this.setState({ isReordering: false, displacement: 0 });
-    this.props.onColumnReorderEnd(ev);
-  };
+    this.checkIfProxyIsDragged();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.columnKey !== this.props.columnKey) {
+      this.checkIfProxyIsDragged();
+    }
```

Comment:
Now that we recycle columns/cells, we need ReorderCell to recheck the presence of drag proxy whenever the column key gets changed.